File: src/mffs/BlockDropDelayedEvent.java
package mffs;

import net.minecraft.block.Block;
import net.minecraft.world.World;
import universalelectricity.core.vector.Vector3;

public class BlockDropDelayedEvent extends DelayedEvent
{
	private Block block;
	private World world;
	private Vector3 position;

	public BlockDropDelayedEvent(int ticks, Block block, World world, Vector3 position)
	{
		super(ticks);
		this.block = block;
		this.world = world;
		this.position = position;
	}

	protected void onEvent()
	{
		this.block.dropBlockAsItem(this.world, this.position.intX(), this.position.intY(), this.position.intZ(), this.position.getBlockMetadata(world), 0);
		this.position.setBlock(this.world, 0);
	}
}

File: src/main/scala/mffs/mobilize/event/BlockDropDelayedEvent.scala
package mffs.mobilize.event

import net.minecraft.block.Block
import net.minecraft.init.Blocks
import net.minecraft.world.World
import universalelectricity.core.transform.vector.Vector3

class BlockDropDelayedEvent(handler: IDelayedEventHandler, ticks: Int, block: Block, world: World, position: Vector3) extends DelayedEvent(handler, ticks)
{
  protected override def onEvent
  {
    if (!this.world.isRemote)
    {
      if (this.position.getBlock(this.world) eq this.block)
      {
        this.block.dropBlockAsItem(this.world, this.position.xi, this.position.yi, this.position.zi, this.position.getBlockMetadata(world), 0)
        this.position.setBlock(this.world, Blocks.air)
      }
    }
  }
}